src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -1099,6 +1121,15 @@ class LicenceService(
     isDueToBeReleasedInTheNextTwoWorkingDays = releaseDateService.isDueToBeReleasedInTheNextTwoWorkingDays(this),
   )
 
+  private fun licenceHolderIsEligible(nomisId: String?): Boolean {
+    if (nomisId == null) {
+      return false
+    }
+
+    val prisoner = prisonerSearchApiClient.searchPrisonersByNomisIds(listOf(nomisId)).first()
+    return eligibilityService.isEligibleForCvl(prisoner)
```

Comment:
could potentially log a warn if submitting an ineligible?

Reply:
Absolutely can do, but as the result of the check failing is to throw an error in all scenarios, I thought we would be able to check the logs for the errors instead. Appreciate this means that if this function is ever used for another purpose, it wouldn't necessarily log

Reply:
well maybe we could change this to `assertCaseEligibile` and throw here?.. which would address that concern and would mean we'd get rid of all the nesting/cruft from above.  

(I don't think we use the term licence holder anywhere else so not sure it makes sense to start using it here?)

Reply:
I agree it's odd, but I wanted to differentiate it from `isEligibleForCvl` which assesses eligibility for cases regardless of if they have a licence already or not. This function is for people who already have licences, and for whom we also don't do HDC checks.
Happy to rename it though and throw the error from here. The only issue is that we'd lose which action (eg edit or submit) caused the error to be thrown, though we should be able to figure that out from the support query and the error trace